### ISSUE TITLE
Sigpy==0.1.23 incompatible with numpy>1.20

### ISSUE THREAD

**Issue by wtclarke** [imr-framework/pypulseq#119]
Hi,

I tried installing this package by cloning and then running `pip install .`, but when running one of the demo scripts `python pypulseq/seq_examples/scripts/write_epi_se_rs.py` the following error was returned:
```
Traceback (most recent call last):
  File "/Users/wclarke/Library/CloudStorage/OneDrive-Nexus365/projects/pps/pypulseq/pypulseq/seq_examples/scripts/write_epi_se_rs.py", line 9, in <module>
    import pypulseq as pp
  File "/Users/wclarke/miniconda3/envs/pps/lib/python3.10/site-packages/pypulseq/__init__.py", line 32, in <module>
    from pypulseq.make_adiabatic_pulse import make_adiabatic_pulse
  File "/Users/wclarke/miniconda3/envs/pps/lib/python3.10/site-packages/pypulseq/make_adiabatic_pulse.py", line 5, in <module>
    from sigpy.mri.rf import hypsec, wurst
  File "/Users/wclarke/miniconda3/envs/pps/lib/python3.10/site-packages/sigpy/__init__.py", line 12, in <module>
    from sigpy import alg, app, config, linop, prox
  File "/Users/wclarke/miniconda3/envs/pps/lib/python3.10/site-packages/sigpy/alg.py", line 8, in <module>
    from sigpy import backend, util
  File "/Users/wclarke/miniconda3/envs/pps/lib/python3.10/site-packages/sigpy/util.py", line 236, in <module>
    def dirac(shape, dtype=np.float, device=backend.cpu_device):
  File "/Users/wclarke/miniconda3/envs/pps/lib/python3.10/site-packages/numpy/__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations. Did you mean: 'cfloat'?
```

The sigpy utils.Dirac function has `np.float` as a default kwarg which is [depreciated as of numpy 1.20](https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations).  This remains the case in the most recent sigpy version https://github.com/mikgroup/sigpy/blob/master/sigpy/util.py#L236. 

I think you either need to pin the numpy version to <1.20 or see if the sigpy devs will update their code. Pinning to an old version of numpy will likely also limit the compatible python versions to <= 3.8. I was able to get everything to run by creating an environment using `conda create -n pps -c conda-forge python=3.8`, modifying the `setup.py` file to `"numpy==1.19.5"` and running `pip install .`.

**To Reproduce**
- conda create -n pps -c conda-forge python=3.10
- git clone https://github.com/imr-framework/pypulseq.git 
- cd pypulseq
- pip install .
- python pypulseq/seq_examples/scripts/write_epi_se_rs.py 

**Expected behavior**
The example script should run.

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: macOS
 - OS Version: 13.2
- `pypulseq` version: Github dev branch - reports as 1.4.0

**Additional context**
I would highly recommend trying out the Github Actions CI to catch these types of issues. I've found it very useful for my [open tools](https://github.com/wtclarke/nifti_mrs_tools/blob/master/.github/workflows/push_pr.yml).

**Comment by wtclarke** [comment]
It looks like there is already some progress on this with sigpy https://github.com/mikgroup/sigpy/issues/123  and https://github.com/mikgroup/sigpy/pull/126

**Comment by bilal-tasdelen** [comment]
@wtclarke Thanks for reporting this. We can update the sigpy version as soon as they merge the fix and release it.

**Comment by curtcorum** [comment]
@wtclarke Nicely reported!
#123 #126
I can confirm that I see it as well. The recent event was the need to use cuda 12.2 for new gpu which triggered other conda/mamba dependencies. There may possibly be a work around set of versions for the environment but much better to update sigpy!

`Linux version 5.4.0-148-generic (buildd@lcy02-amd64-056) (gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)) #165~18.04.1-Ubuntu SMP Thu Apr 20 01:14:06 UTC 2023`

```
cuda-toolkit              12.2.0                        0    nvidia
cupy                      12.1.0           py39hc7c1505_0    conda-forge
numpy                     1.25.1           py39h6183b62_0    conda-forge
python                    3.9.0           hffdb5ce_5_cpython    conda-forge
sigpy                     0.1.25             pyha21a80b_1    conda-forge
```

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[19], line 2
      1 #########################################################
----> 2 import sigpy as sp
      3 import sigpy.mri as mr
      4 import sigpy.plot as pl

File ~/miniconda3/envs/cuda_12_2/lib/python3.9/site-packages/sigpy/__init__.py:12
      1 """The core module contains functions and classes for signal processing.
      2 
      3 SigPy provides simple interfaces to commonly used signal processing functions,
   (...)
      9 
     10 """
     11 from .version import __version__ # noqa
---> 12 from sigpy import alg, app, config, linop, prox
     14 from sigpy import (backend, block, conv, interp,
     15                    fourier, pytorch, sim, thresh,
     16                    util, wavelet)
     17 from sigpy.backend import *  # noqa

File ~/miniconda3/envs/cuda_12_2/lib/python3.9/site-packages/sigpy/alg.py:8
      6 import numpy as np
      7 import sigpy as sp
----> 8 from sigpy import backend, util
     11 class Alg(object):
     12     """Abstraction for iterative algorithms.
     13 
     14     The standard way of using an :class:`Alg` object, say alg, is as follows:
   (...)
     45 
     46     """

File ~/miniconda3/envs/cuda_12_2/lib/python3.9/site-packages/sigpy/util.py:236
    231     output[slc] = input
    233     return output
--> 236 def dirac(shape, dtype=np.float, device=backend.cpu_device):
    237     """Create Dirac delta.
    238 
    239     Args:
   (...)
    245         array: Dirac delta array.
    246     """
    248     device = backend.Device(device)

File ~/miniconda3/envs/cuda_12_2/lib/python3.9/site-packages/numpy/__init__.py:313, in __getattr__(attr)
    308     warnings.warn(
    309         f"In the future `np.{attr}` will be defined as the "
    310         "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    312 if attr in __former_attrs__:
--> 313     raise AttributeError(__former_attrs__[attr])
    315 if attr == 'testing':
    316     import numpy.testing as testing

AttributeError: module 'numpy' has no attribute 'float'.
`np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
```

**Comment by schuenke** [comment]
I am currently preparing a pypulseq hands-on session for this friday and installing PyPulseq is really a pain in the a**

Can we please push a hotfix to master and release a version 1.40.post1?

The only thing we have to do is change the dependencies from `"sigpy==0.1.23"` to `"sigpy>=0.1.25"`


We can still discuss to remove the sigpy dependency completely in the future, but for now we need a quick solution IMO!

**Comment by wtclarke** [comment]
This is fixed with the merge of #148
